src/functions.jl

```diff
@@ -174,24 +166,43 @@ for (T, S) in [
     end
 end
 
-Base.filter(f, A::NamedDimsArray{L,T,1}) where {L,T} = NamedDimsArray(filter(f, parent(A)), L)
-Base.filter(f, A::NamedDimsArray{L,T,N}) where {L,T,N} = filter(f, parent(A))
-
+function Base.filter(f, A::NamedDimsArray{L, T, 1}) where {L, T}
+    return NamedDimsArray(filter(f, parent(A)), L)
+end
+Base.filter(f, A::NamedDimsArray{L, T, N}) where {L, T, N} = filter(f, parent(A))
 
 # We overload collect on various kinds of `Generators` so that that can keep names.
 function Base.collect(x::Base.Generator{<:NamedDimsArray{L}}) where {L}
     data = collect(Base.Generator(x.f, parent(x.iter)))
     return NamedDimsArray(data, L)
 end
 
-function Base.collect(x::Base.Generator{<:Iterators.Enumerate{<:NamedDimsArray{L}}}) where {L}
+function Base.collect(
+    x::Base.Generator{<:Iterators.Enumerate{<:NamedDimsArray{L}}},
+) where {L}
     data = collect(Base.Generator(x.f, enumerate(parent(x.iter.itr))))
     return NamedDimsArray(data, L)
 end
 
-Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:NamedDimsArray,Vararg{Any}}}}) = collect_product(x)
-Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:Any,<:NamedDimsArray,Vararg{Any}}}}) = collect_product(x)
-Base.collect(x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:NamedDimsArray,<:NamedDimsArray,Vararg{Any}}}}) = collect_product(x)
+function Base.collect(
+    x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{<:NamedDimsArray, Vararg{Any}}}},
+)
+    return collect_product(x)
+end
+function Base.collect(
+    x::Base.Generator{<:Iterators.ProductIterator{<:Tuple{
+        <:Any, <:NamedDimsArray, Vararg{Any},
+    }}},
```

Comment:
I am not sure about this

Reply:
See #138

Reply:
it's a consequence of the original code ignoring line limits, and the formatter making the best attempt at a tough job (it did fine in my opinion), but i fixed it up to something nicer in #138